geoips/commandline/geoips_list.py

```diff
@@ -0,0 +1,565 @@
+        args: Argparse Namespace()
+            - The Argument Namespace for GeoipsListInterfaces Sub-Command
+        """
+        package_name = args.package_name
+        # Flag representing whether or not we want to list what's implemented or
+        # what's available.
+        implemented = args.implemented
+        if package_name != "all" or (implemented and package_name):
+            # If you're listing a certain package, implemented is implied. If it's
+            # 'all' packages, make sure implemented as added if we are going to list in
+            # that fasion.
+            self.list_implemented_interfaces(package_name)
+        else:
+            # Otherwise just list off available interfaces.
+            self.list_available_interfaces()
+
+    def list_available_interfaces(self):
+        """List the available interface[s] within [a] GeoIPS Package[s]".
+
+        Printed to Terminal
+        -------------------
+        out_array: 2D Array of Strings
+            - Absolute Path
+            - Docstring
+            - GeoIPS Package
+            - Interface Type
+            - Interface Name
+            - Supported Families
+        """
+        interface_data = []
+        for interface_name in interfaces.__all__:
+            interface = getattr(interfaces, interface_name)
+            interface_entry = [
+                "geoips",
+                interface.interface_type,
+                interface_name,
+                ",\n".join(interface.supported_families),
+                interface.__doc__.split("\n")[0],
+                str(
+                    resources.files("geoips")
+                    / f"interfaces/{interface.interface_type}/{interface.name}.py"
+                ),
+            ]
+            interface_data.append(interface_entry)
+        headers = [
+            "GeoIPS Package",
+            "Interface Type",
+            "Interface Name",
+            "Supported Families",
+            "Docstring",
+            "Absolute Path",
+        ]
+        print("-" * len("GeoIPS Interfaces"))
+        print("GeoIPS Interfaces")
+        print("-" * len("GeoIPS Interfaces"))
+        print(
+            tabulate(
+                interface_data,
+                headers=headers,
+                tablefmt="rounded_grid",
+                maxcolwidths=self.terminal_width // len(headers),
+            )
+        )
+
+    def list_implemented_interfaces(self, package_name):
+        """List the implemented interface[s] within [a] GeoIPS Package[s].
+
+        Ie. search through all, or an individual package and list off what has been
+        implemented in such package[s].
+
+        Printed to Terminal
+        -------------------
+        out_array: 2D Array of Strings
+            - GeoIPS Package
+            - Interface Type
+            - Interface Name
+
+        Parameters
+        ----------
+        package_name: str
+            - The GeoIPS Package name whose scripts you want to list.
```

Comment:
"Implemented" here means that a package contains plugins for a particular interface, correct? I think we probably need some different language around this to make it more clear.

Reply:
Yes, that's exactly what it means. Do you have any ideas of what you'd like to call this instead? This might cause a decent amount of conflicts. It may be better to handle this in a future PR (we can create an issue for this), but we can handle it in this one if you'd prefer @jsolbrig.

Reply:
I'll create an issue to consider this. It's fine for this PR.

Reply:
Created https://github.com/NRLMMD-GEOIPS/geoips/issues/586